### PR TITLE
Implement Display + Error for error types

### PR DESCRIPTION
**Pull request by zoeyfyi**
Also fixes the 2 clippy lints.

### CODE REVIEW

File: gladis/src/lib.rs [hunk 1]
```diff
@@ -142,3 +170,4 @@ pub trait Gladis {
 #[cfg(feature = "derive")]
 #[doc(hidden)]
 pub use gladis_proc_macro::Gladis;
+use std::{error::Error, fmt::Display};
```
Comment (gagath) [inline gladis/src/lib.rs]: I am confused why this is at the end of the file and not at the top, just before `use gtk;`. Can you move it up?

File: gladis/src/lib.rs [hunk 2]
```diff
@@ -44,11 +44,39 @@ pub struct NotFoundError {
     pub typ: String,
 }
 
+impl Display for NotFoundError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "Identified {} of type {} was not found",
```
Comment (gagath) [inline gladis/src/lib.rs]: typo: "Identified" -> "Identifier".
Also, do not use uppercase here, because this will lead `GladisError::fmt` to return strange case:

> Not found error: Identifier foo of type gtk::Label was not found

It should be "identifier" all in small caps.

**Review by gagath — CHANGES_REQUESTED** [review]
Thanks for your contribution! Some points have to be addressed before merge.

Also, do you think we can introduce some unit tests to verify the formatting of the errors? They can be placed at the very end of the file with the following construct:

```
#[cfg(test)]
mod tests {
   // ...
}
```

https://doc.rust-lang.org/rust-by-example/testing/unit_testing.html

File: gladis/src/lib.rs (hunk 2 above)
Comment (zoeyfyi) [reply gladis/src/lib.rs]: Should it be all lowercase? ([guidelines](https://rust-lang.github.io/api-guidelines/interoperability.html#error-types-are-meaningful-and-well-behaved-c-good-err)). Otherwise it would look strange if the error was wrapped by a consumer of the crate, i.e `Gladis error: Not found error: identifier foo of type gtk::Label was not found`.

So should it be `not found error: identifier foo of type gtk::Label was not found`

File: gladis/src/lib.rs (hunk 1 above)
Comment (zoeyfyi) [reply gladis/src/lib.rs]: Sure, this is just where rust-analyzer choose to insert it. There is no `use gtk;` in this file so I just moved it to the standard location bellow the module documentation, is that ok?

Comment (gagath) [reply gladis/src/lib.rs]: You did well, I mixed with another file.

File: gladis/src/lib.rs (hunk 2 above)
Comment (gagath) [reply gladis/src/lib.rs]: All lowercase is better I think, to avoid this kind of composition issue.

**Review by gagath — APPROVED** [review]

File: gladis/src/lib.rs [hunk 3]
```diff
@@ -30,25 +30,51 @@
...
 //!
-//! fn main() {
-//!     gtk::init().unwrap();
-//!     let _ui = Window::from_string(GLADE_SRC).unwrap();
-//! }
+//! gtk::init().unwrap();
+//! let _ui = Window::from_string(GLADE_SRC).unwrap();
 //! ```
 
-use gtk;
```
Comment (gagath) [inline gladis/src/lib.rs:39]: Why is this removed? gtk is needed for the Gladis trait definition

**Review by gagath — CHANGES_REQUESTED** [review]

File: gladis/src/lib.rs (hunk 3 above)
Comment (zoeyfyi) [reply gladis/src/lib.rs:39]: https://rust-lang.github.io/rust-clippy/master/index.html#single_component_path_imports

Comment (zoeyfyi) [reply gladis/src/lib.rs:39]: https://rust-lang.github.io/rust-clippy/master/index.html#single_component_path_imports

Comment (gagath) [reply gladis/src/lib.rs:39]: My bad, thanks!

**Review by gagath — APPROVED** [review]

**Comment by gagath** [comment]
This PR is present in release v0.4.1: https://crates.io/crates/gladis/0.4.1
